src/Npgsql.NodaTime/Internal/PeriodConverter.cs

```diff
@@ -17,6 +24,15 @@ protected override Period ReadCore(PgReader reader)
         var days = reader.ReadInt32();
         var totalMonths = reader.ReadInt32();
 
+        if (microsecondsInDay == long.MaxValue && days == int.MaxValue && totalMonths == int.MaxValue)
+            return _dateTimeInfinityConversions
+                ? Period.MaxValue
+                : throw new InvalidCastException(NpgsqlNodaTimeStrings.CannotReadInfinityValue);
```

Comment:
Hmm, am wondering... These were valid, non-infinity values before PG17, right? Is this a breaking change? What happens with databases that already have these values from before PG17, they suddenly start being interpreted as inifinity?

We should do a breaking note if that's the case.

Reply:
Discussed offline. So, if a user upgrades to PG17, their value will remain the same. The issue is that what user will actually get heavily depends on multiple things.  For starters, if a user reads the value via simple query protocol (or extended with text conversion) or the value is converted to text on PG's side, then they'll get infinity. Otherwise, if they're using extended query protocol with binary values, then it's completely up to a client. PSQL always requests values in text, so in it's case whether it's infinity or the actual value will depend on postgres' version.
As for us, we couldn't read this kind of values before because we always failed during `Normalize` call.

Reply:
> (BTW did we support writing them? 😈 )

Oh, that's a funny thing. It depends 🐝 
Before 8.0.5 (specifically #5867), we never normalized our writes. So we did write them. Kinda. The problem is, due to every value being max, most of our multiplications we do to convert to PG representation will result in 0 due to overflow. So in the end we do write something but you don't get the exact value you were writing. With 8.0.5, we now normalize beforehand, which does result in overflow, the same as read side.

Reply:
Thinking a bit more, it does sound like a bug where due to silent overflow we accepted invalid values. We should probably fix that...